### PR TITLE
esn-frontend-mailto#13: Corrected the behaviour of the composer's "delete" button in the Mailto SPA

### LINKED ISSUE #13: Correct the behaviour of the composer's "delete" button in the Mailto SPA
I'm not really sure myself what should be the behaviour of the composer's "delete" button in the Mailto SPA. Should we close the Mailto app when the button is clicked (in the Inbox SPA, the composer is closed when the button is clicked)?

### PR DESCRIPTION
**Pull request by huy-ta**
Partially resolves #13. Please also refer to http://github.com/OpenPaaS-Suite/esn-frontend-inbox/pull/89.

Demo: 

[Demo-Mailto-Delete-Draft](https://streamable.com/3z633s)

**Notes:**

- The title of the button is not showing in the demo because my screen recorder, together with the screenshot app, doesn't capture it for some reason.
- In case you wonder why the window is not closing, `window.close()` only works when the window is opened by another (parent) window. If you navigate to the page by manually entering the URL, `window.close()` will not work. It's an intended behaviour of browsers: https://developer.mozilla.org/en-US/docs/Web/API/Window/close. Sadly we have no way to detect if `window.close()` is going to work or not to provide some kind of proper UI feedback.

### CODE REVIEW

**Comment by chamerling** [comment]
Can we have a video also?

**Comment by huy-ta** [comment]
> Can we have a video also?

Yes I'm recording it.

File: src/app/components/mailto-page/mailto-page.controller.js [hunk 1]
```diff
@@ -23,29 +20,25 @@ angular
...
     $scope.$on(MAILTO_MAIL_STATUS_EVENTS.UPDATED, (event, newStatus) => {
-      if (newStatus !== self.MAILTO_MAIL_STATUSES.SENDING) {
+      if (newStatus === self.MAILTO_MAIL_STATUSES.SENT || newStatus === self.MAILTO_MAIL_STATUSES.FAILED) {
         self.status = self.MAILTO_MAIL_STATUSES.TRANSITION;
 
         // This has to be 500ms to be in sync with the animation time defined in the LESS file.
-        $timeout(() => {
+        return $timeout(() => {
           self.status = newStatus;
         }, 500);
```
Comment (tuanlc) [inline src/app/components/mailto-page/mailto-page.controller.js:31]:
```suggestion
        $timeout(() => {
          self.status = newStatus;
        }, 500);
        
        return;
```

**Comment by tuanlc** [comment]
When the `Reopen` button is disabled, it is better if there is a `title` to explain why to users when they hover to the button.

**Comment by huy-ta** [comment]
> When the `Reopen` button is disabled, it is better if there is a `title` to explain why to users when they hover to the button.

Yes that's a good idea.

File: src/app/components/mailto-page/mailto-page.controller.js (hunk 1 above)
Comment (huy-ta) [reply src/app/components/mailto-page/mailto-page.controller.js]: Fixed.

**Review by chamerling — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

**Review by tuanlc — APPROVED** [review]

**Pull request merged**